eth/protocols/eth/celo_peer.go

```diff
@@ -0,0 +1,24 @@
+package eth
+
+import (
+	"bytes"
+
+	"github.com/celo-org/celo-blockchain/p2p"
+	"github.com/celo-org/celo-blockchain/rlp"
+)
+
+// Send writes an RLP-encoded message with the given code.
+// data should encode as an RLP list.
+func (p *Peer) Send(msgcode uint64, data []byte) error {
+	raw, err := rlp.EncodeToBytes(data)
+	if err != nil {
+		return err
+	}
+	return p.SendDoubleEncoded(msgcode, raw)
+}
+
+// SendDoubleEncoded sends an rlp encoded (twice) message with the given code.
+func (p *Peer) SendDoubleEncoded(msgcode uint64, data []byte) error {
```

Comment:
I understand the desire to emphasize the fact that the encoding is performed twice but the implementation of the function doesn't have anything with encoding at all it which is misleading. Logically, `SendDoubleEncoded` should become `Send` and `Send` should become `SendEncoded`. And in the [calling code](https://github.com/celo-org/celo-blockchain/pull/2175/files#diff-8e3ffc7b008ae6e501c7aa2cb2eed2db1fbef6586d767f59b1209fef26ad5ac1R121) I would leave a comment explaining the incidental (hopefully?) double encoding.
Otherwise, I like the change.